packages/fuels-core/src/parameters.rs

```diff
@@ -14,25 +15,30 @@ pub struct TxParameters {
 pub struct CallParameters {
     pub amount: u64,
     pub asset_id: AssetId,
-    pub gas_forwarded: u64,
+    gas_forwarded: Option<u64>,
 }
 
 impl CallParameters {
     pub fn new(amount: Option<u64>, asset_id: Option<AssetId>, gas_forwarded: Option<u64>) -> Self {
         Self {
-            amount: amount.unwrap_or(0),
+            amount: amount.unwrap_or(DEFAULT_CALL_PARAMS_AMOUNT),
             asset_id: asset_id.unwrap_or(BASE_ASSET_ID),
-            gas_forwarded: gas_forwarded.unwrap_or(DEFAULT_FORWARDED_GAS),
+            gas_forwarded,
         }
     }
+
+    /// Returns gas_forwarded if set or default if None
+    pub fn gas_forwarded_or(&self, default: u64) -> u64 {
```

Comment:
Thinking out loud, no action needed:

Maybe we could have just returned the option and had the users call `unwrap_or` on it. Or just have this as an exception and not do this too often. Wouldn't like to do this for all options out there.

With the 'return Option approach' maybe the comment wouldn't be warranted since an option implies that something might be missing and thus you've decided on the `or` part.

Technically somebody might question, reading only the name of the method, what conditions need to be met for this method to not give the gas forwarded but rather the default passed in. The comment was needed to further explain that the gas forwarded might not have been set.

Reply:
It was bit of desperate attempt to provide more context as to what is happening since the logic that defaults to gas_limit is deeply embedded into the call script generation.
But looking at it now, it doesn't achieve much so we might as well go for `unwrap_or `